File: cpp/leetcode-minimumchangealternate.cpp
#include <string>
using namespace std;
class Solution {
public:
    int minOperations(string s) {
        int changes01 = 0;  
        int changes10 = 0; 


        for (int i = 0; i < s.length(); i++) {
            if (i % 2 == 0 && s[i] != '0') {
                changes01++;
            } else if (i % 2 == 1 && s[i] != '1') {
                changes01++;
            }
        }

        for (int i = 0; i < s.length(); i++) {
            if (i % 2 == 0 && s[i] != '1') {
                changes10++;
            } else if (i % 2 == 1 && s[i] != '0') {
                changes10++;
            }
        }

        return min(changes01, changes10);
    }
};

File: py/leetcode-minimumchangealternate.py
class Solution:
    def minOperations(self, s: str) -> int:
        changes_01 = 0  
        changes_10 = 0  

        for i in range(len(s)):
            if i % 2 == 0 and s[i] != '0':
                changes_01 += 1
            elif i % 2 == 1 and s[i] != '1':
                changes_01 += 1

        for i in range(len(s)):
            if i % 2 == 0 and s[i] != '1':
                changes_10 += 1
            elif i % 2 == 1 and s[i] != '0':
                changes_10 += 1

        return min(changes_01, changes_10)